com.unity.render-pipelines.high-definition/Runtime/Sky/PhysicallyBasedSky/PhysicallyBasedSkyRenderer.cs

```diff
@@ -318,26 +318,23 @@ unsafe struct ShaderVariablesPhysicallyBasedSky
             public float _AerosolDensityFalloff;
             public float _AerosolScaleHeight;
 
-            public Vector3 _AirSeaLevelExtinction;
-            public float _AerosolSeaLevelExtinction;
+            public Vector4 _AirSeaLevelExtinction;
+            public Vector4 _AirSeaLevelScattering;
+            public Vector4 _AerosolSeaLevelScattering;
+            public Vector4 _GroundAlbedo;
+            public Vector4 _PlanetCenterPosition; // Not used during the precomputation, but needed to apply the atmospheric effect
+            public Vector4 _HorizonTint;
+            public Vector4 _ZenithTint;
```

Comment:
We can't use Vector3 at all? GenerateHLSL generates broken code?

Reply:
Nothing wrong with Vector3, but when the generated hlsl is converted to either MSL (Metal Shading Language) or SPIR-V, your struct will actually have a different layout if you use it (in Metal it will have a different size as well as a different layout).

Reply:
So it's not broken, then? This is just for clarity? We can't really expect auto-generated structs to be "broken" and then manually patch them in each case.

Reply:
No, the existing constant buffer layout is totally broken. We get Metal validation errors from this because we've declared a struct that we think is one size/layout (the way it would layout for DX11), but really it's another.  So the buffer we create to fulfill that binding requirement is the wrong size, and Metal screams at us.  Even without the validation errors, the data is now in the wrong location, so the rendering is wrong.